doc/RFCs/APInotification.md

```diff
@@ -0,0 +1,245 @@
+    to: message-component  
+    id: xxxxx-xxxxx-xxxxx  
+    type: event  
+    timestamp: 1518691674517  
+    body:  
+      emailReceived: xxxxxx-xxxxx-xxxxx // uuid of new email  
+  - from: email-facility  
+    to: message-component  
+    id: xxxxx-xxxxx-xxxxx  
+    type: event  
+    timestamp: 1518691674517  
+    body:  
+      emailReceived: xxxxxx-xxxxx-xxxxx // uuid of new email  
+  - from: notification-center  
+    to: browser-notification  
+    id: xxxxx-xxxxx-xxxxx  
+    type: teaser  
+    timestamp: 1518691674517  
+    body: Pour améliorer votre PI vous pourriez faire ceci…  
+```
+
+### Protocole :
+
+Le frontend fait des requêtes HTTP (GET ou DELETE) sur /api/v2/notifications.
+
+Le serveur répond avec un payload en JSON et ferme la connexion.
+
+### Endpoint et API :
+
+#### endpoint : `…/api/v2/notifications`
+
+#### méthodes : 
+
+`GET` pour obtenir des notifications en file d'attente :
+
+- sans params dans la query, toutes les notifications de la file d'attente sont retournées dans un seul document.
+- 2 params optionnels : `from` et `to` pour filtrer les notifications par timestamp.
+
+`DELETE` pour supprimer des notifications de la file d'attente
+
+- 1 param obligatoire : `until` pour indiquer jusqu'à quel timestamp les notifications doivent être supprimées.
+
+#### spécifications (swagger) : voir [/src/backend/defs/rest-api/paths/notifications.yaml](/src/backend/defs/rest-api/paths/notifications.yaml)
+
+### Principe de fonctionnement :
+
+#### Backend :
+
+Le backend gère une pile de notifications pour chaque utilisateur.
+
+- De nouvelles notifications sont créées et ajoutées à la pile par le backend en fonction d'actions de l'utilisateur (création d'un contact par ex.), d'événements extérieurs (arrivée d'un nouvel email par ex.),de traitements batch, etc.
+
+Chaque notification à une durée de vie (TTL) prédéfinie dans la pile, en fonction de son niveau d'importance.
+
+Par conséquent, la pile de notifications est automatiquement purgée.
+
+#### Frontend:
+
+Lors d'un call `GET` du frontend, la pile de toutes les notifications en instance est envoyée.
+
+Le frontend peut refaire un `GET` xx secondes plus tard, il aura la même pile de notifications +/- les notifications arrivées entre-temps et celles qui ont expirées.
+
+Le frontend peut refaire un `GET` xx secondes plus tard en ajoutant le param from=xxxxxxx pour ne recevoir que les dernières notifications arrivées depuis son dernier call (ou depuis le timestamp de la notification la plus récente qu'il a reçu lors du dernier call)
+
+Le frontend peut explicitement purger la pile de notifications côté backend en faisant un call `DELETE` avec le param `until`
+
+### Modèle de données :
+
+La table des notifications est stockées dans cassandra.
+
+#### Chaque `INSERT` se fera avec la directive `USING TTL xx` pour gérer le nettoyage automatique de la table
+
+```sql  
+CREATE TABLE user_notification (  
+user_id uuid,  
+timestamp timestamp,  
```

Comment:
device_id is may be needed to identity for which device the notification is

Reply:
But, a notification is not always for a specific device.  
For example:  if a new email arrived while an user is disconnected, a notification is queued. But, we don't know yet which device the user will use to reconnect. So the notification is « for first device ». Isn't it ?

Reply:
Yes you're right, notification will be seen by all actives devices by default.

Reply:
Yes you're right. May they will have notification dedicated to specific devices, but later, I ensure at this time about that need.